### PR TITLE
Set environment variables USER and LOGNAME for CRI jobs

### PR DESCRIPTION
**Pull request by koct9i**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

**Comment by koct9i** [comment]
YT CPP SDK fails if cannot get username 

erminate called after throwing an instance of 'yexception'
  what():  /root/.conan2/p/b/twix-3d94e104ee873/b/ytsaurus/yt/cpp/mapreduce/interface/config.cpp:286: Cannot get user name: (Success) /root/.conan2/p/b/twix-3d94e104ee873/b/ytsaurus/util/system/user.cpp:52:  getpwuid_r failed

But it could get from environment

https://github.com/ytsaurus/ytsaurus/blob/main/util/system/user.cpp#L15

There are two more-or-less sandard unix environment variables for that:
https://github.com/util-linux/util-linux/blob/master/login-utils/login.c#L1222

So, let's set them both.

This code-path which passes environment around (exec-node -> job-proxy -> yt-exec -> user-job) is ugly as hell. Let's leave cleaing this mess for future us.

File: yt/yt/server/job_proxy/environment.cpp [hunk 1]
```diff
@@ -819,6 +820,12 @@ class TCriUserJobEnvironment
     : public IUserJobEnvironment
 {
 public:
+    TCriUserJobEnvironment() {
```
Comment (gritukan) [inline yt/yt/server/job_proxy/environment.cpp:823]: { should be on the next line

File: yt/yt/server/job_proxy/environment.cpp [hunk 2]
```diff
@@ -819,6 +820,12 @@ class TCriUserJobEnvironment
...
+	TString username = ::GetUsername();
```
Comment (gritukan) [inline yt/yt/server/job_proxy/environment.cpp:824]: auto?

File: yt/yt/server/node/exec_node/job_environment.cpp [hunk 3]
```diff
@@ -1095,6 +1096,17 @@ class TCriJobEnvironment
         }
         spec->Credentials.Gid = ::getgid();
 
+        if (!spec->Environment.contains("USER")) {
+            TString username;
```
Comment (gritukan) [inline yt/yt/server/node/exec_node/job_environment.cpp:1100]: I'm not sure if I got it correctly. Why isn't it sufficient just to add environment variables in TCriUserJobEnvironment::TCriUserJobEnvironment() like you did and we have to do it twice?

Comment (koct9i) [reply yt/yt/server/node/exec_node/job_environment.cpp:1100]: Here we fill environemtn for job-proxy.
TCriUserJobEnvironment fills environement for user-job.

Comment (gritukan) [reply yt/yt/server/node/exec_node/job_environment.cpp:1100]: Oh, I see.

**Review by gritukan — APPROVED** [review]

**Comment by koct9i** [comment]
Merge conflict is resolved. Could be merged now.

**Pull request closed without merging**